### PR TITLE
Enable multi-select on cohort rule

### PR DESCRIPTION
**Pull request by eanders**
This is the Cohort equivalent of this: https://github.com/greenriver/boston-cas/pull/464

### CODE REVIEW

File: spec/models/rules/active_in_cohort_spec.rb [hunk 1]
```diff
@@ -52,7 +58,36 @@
...
+      end
+      it 'does not contain Sue' do
+        expect(clients_that_fit_multi.ids).to_not include sue.id
+      end
+      it 'does not contain Zelda' do
+        expect(clients_that_fit_multi.ids).to_not include zelda.id
+      end
+      it 'does not contain Zelda' do
+        expect(clients_that_fit_multi.ids).to_not include zelda.id
+      end
```
Comment (kwiliarty) [inline spec/models/rules/active_in_cohort_spec.rb]: I think these lines are identical to the previous three.

File: app/models/rules/active_in_cohort.rb [hunk 2]
```diff
@@ -19,19 +20,24 @@ def available_cohorts
...
+    # Note, we need to jump through a few hoops since variable wasn't designed as a json field
+    available_cohorts.select { |id, _| id.to_s.in?(value_as_array(value)) }.map(&:last).join(' or ') || value
   end
 
-  def clients_that_fit(scope, requirement, opportunity)
+  def clients_that_fit(scope, requirement, _opportunity)
     if Client.column_names.include?(:active_cohort_ids.to_s)
       if requirement.positive
-        where = 'active_cohort_ids @> ?'
+        where = 'active_cohort_ids @> ANY(ARRAY [?]::jsonb[])'
```
Comment (kwiliarty) [inline app/models/rules/active_in_cohort.rb:30]: I'm not familiar with this syntax. Can you tell me what it's doing?

Comment (eanders) [reply app/models/rules/active_in_cohort.rb:30]: The question mark will be replaced with the value when dropped into the where statement down below.  They @> is postgres json query syntax for "in" (https://www.postgresql.org/docs/9.5/functions-json.html.)  So, force the string provided into an postgres array, and see if any of the provided values are in the active_cohort_ids column.

Comment (kwiliarty) [reply app/models/rules/active_in_cohort.rb:30]: It was the `@>` that was new to me. Thanks!

**Review by kwiliarty — APPROVED** [review]
This looks good to me.

**Pull request merged**